pkg/scm/scm.go

```diff
@@ -16,41 +15,48 @@ import (
 // DownloaderForSource determines what SCM plugin should be used for downloading
 // the sources from the repository.
 func DownloaderForSource(s string) (build.Downloader, string, error) {
-	// If the source is using file:// protocol but it is not a GIT repository,
+	// If the source exists locally  but it is not a GIT repository,
 	// trim the prefix and treat it as a file copy.
-	if strings.HasPrefix(s, "file://") && !isLocalGitRepository(s) {
-		s = strings.TrimPrefix(s, "file://")
+	fileCopy := false
+	if doesExist(s, true) && !isLocalGitRepository(s) {
+		fileCopy = true
 	}
 
-	// If the source is file:// protocol and it is GIT repository, but we don't
+	// If the source exists locally and it is GIT repository, but we don't
 	// have GIT binary to fetch it, treat it as file copy.
-	if strings.HasPrefix(s, "file://") && !hasGitBinary() {
-		s = strings.TrimPrefix(s, "file://")
+	if doesExist(s, true) && !hasGitBinary() {
+		fileCopy = true
 	}
 
-	// If the source is valid GIT protocol (file://, git://, git@, etc..) use GIT
-	// binary to download the sources
-	if g := git.New(); g.ValidCloneSpec(s) {
-		return &git.Clone{g, util.NewFileSystem()}, s, nil
+	g := git.New()
+	if !g.ValidCloneSpecRemoteOnly(s) {
+		fileCopy = true
 	}
 
-	// Convert relative path to absolute path.
-	if !strings.HasPrefix(s, "/") {
-		if absolutePath, err := filepath.Abs(s); err == nil {
-			s = absolutePath
+	if fileCopy {
+		if doesExist(s, true) {
+			if !isLocalGitRepository(s) {
+				// since not using git, any resulting URLs need to be explicit with file:// protocol specified
+				s = "file://" + s
```

Comment:
ah ...late night coding ... the boolean param on doesExist was part of an attempt to deal with file:// either being there or not, without multiple checks for the prefix, but did not complete the job;

i'll rework and update